ert_shared/ensemble_evaluator/ensemble/prefect.py

```diff
@@ -57,14 +57,13 @@ async def _eq_submit_job(self, script_filename):
 
 
 def _get_executor(custom_port_range, name="local"):
-    _, port, _ = find_available_port(custom_range=custom_port_range, reuse_addr=True)
     if name == "local":
         cluster_kwargs = {
             "silence_logs": "debug",
-            "scheduler_options": {"port": port},
         }
         return LocalDaskExecutor(**cluster_kwargs)
     elif name == "lsf":
+        _, port, _ = find_available_port(custom_range=custom_port_range)
```

Comment:
I thought the reason for having the `will_close_then_reopen_socket`, was that it was needed here?

Reply:
I thought so too initially but while writing the test-suite I figured out that the desired behaviour can be obtained without the flag. But keep reading please...

Setting the flag corresponds to pattern in the test named `test_reuse_passive_close_ok_ok`, i.e. obtain the initial port/socket with flag=true, then close socket without using it. Observe that we get the same behaviour when **not** setting the flag, demonstrated in `test_def_passive_live_nok_nok_close_ok_ok` (read its executive summary).

However, when writing this commnent, I realize that both tests explicitly close the socket while ERT relies on that to happen automatically. According to [Python documentation](https://docs.python.org/3.8/library/socket.html#socket-objects) :

> Sockets are automatically closed when they are garbage-collected, but it is
> recommended to close() them explicitly, or to use a with statement around them.

Hence, this code

` _, port, _ = find_available_port(custom_range=custom_port_range)
`

relies on GC kicking in **before** Dask tries to bind the port, which probably not happens from time to time. Moreover, according to the test `test_reuse_passive_live_linux_nok_ok`, if the socket is not closed (GCd in our case) Dask will fail to bind the port on Linux (unless it sets SO_REUSEADDR, which it doesn't). IMO this is a likely source of flakiness.

So I'll probably just replace ERT code with

```
 _, port, s = find_available_port(custom_range=custom_port_range)
s.close()
```

following the pattern in `test_def_passive_live_nok_nok_close_ok_ok` and getting the behaviour we want by (deterministically) closing the socket.

_**Note that nothing above reserves the port until it is bound by Dask, which is an entirely different issue. Any other process may hijack the port between us closing the socket and Dask re-binding the port. Fixing this requires Dask to accept an already bound socket, a port-range in which to find the port, or alternatively a socket-factory or similar.**_

Thoughts?

Reply:
Regarding gc. Might be worthwhile making a test(s) that explicitly invokes the gc to verify the desired behavior.

Reply:
> following the pattern in test_def_passive_live_nok_nok_close_ok_ok and getting the behaviour we want by (deterministically) closing the socket.

Will this allow us to rebind the socket immediately? :thinking:

Reply:
> Will this allow us to rebind to the same port immediately? 🤔

AFICS, yes. Please refer to testcode (pointing out any mistakes, of course  😄  )

Reply:
> Regarding gc. Might be worthwhile making a test(s) that explicitly invokes the gc to verify the desired behavior.

I'm inclined to trust Python on this, but I'll give it a shot.

Reply:
Mmmm...  that was an interesting exercise... 😬 

Looks like GC is not the crucial part but rather refcounting. I added `test_gc_closes_socket` which may or may not provide insight or value. Didn't figure out how to mock close() because that method belongs to the superclass `_socket.socket` which is a built-in or extension type, apparently.

Reply:
Alright, then I am fine with merging this!

Reply:
Since I don't have the overview of use of Prefect - could you point out which test(s) are most likely to be affected by this change?